### PR TITLE
integration tests: add option to use pprof profiler

### PR DESCRIPTION
**Pull request by dougfort**
Signed-off-by: dougfort <doug.fort@greymatter.io>

This patch enables the use of the go language pprof profiler in integration tests.

### CODE REVIEW

File: pkg/test/main/README.md
Comment (alecholmez) [inline pkg/test/main/README.md:51]: I'm guessing this just gets written to the root of the project?

File: pkg/test/main/README.md
Comment (alecholmez) [inline pkg/test/main/README.md:52]: Maybe we can have a make target so this is easy?

```make
make analyze
```

File: pkg/test/main/main.go
Comment (alecholmez) [inline pkg/test/main/main.go:146]: I'd follow the previous naming patterns and do camelCase instead of snake case. 

```suggestion
flag.BoolVar(&pprofEnabled, "pprof", false, "Enable use of the pprof profiler")
```

File: pkg/test/main/main.go
Comment (alecholmez) [inline pkg/test/main/main.go:278]: maybe remove this whitespace?

File: pkg/test/main/main.go
Comment (alecholmez) [inline pkg/test/main/main.go:283]: Do you think it makes sense to have multiple files created out of this? I'm not entirely sure how pprof generates its artifacts but would it be possible to have everything written to one file?

**Review by alecholmez — CHANGES_REQUESTED** [review]
Just some minor comments, I think this is a great start to the full e2e benchmark. Now we've got some sort of profiling data so we can start generating artifacts based on difference scenarios we create

File: pkg/test/main/main.go
Comment (dougfort) [reply pkg/test/main/main.go:278]: roger

File: pkg/test/main/README.md
Comment (dougfort) [reply pkg/test/main/README.md:52]: I don't want to mess with the makefile yet. 

Note that they have multiple targets:
```
integration.ads: $(BINDIR)/test $(BINDIR)/upstream
	env XDS=ads build/integration.sh

integration.xds: $(BINDIR)/test $(BINDIR)/upstream
	env XDS=xds build/integration.sh

etc
```
I agree that we need a more convenient way to run this.

File: pkg/test/main/main.go
Comment (dougfort) [reply pkg/test/main/main.go:283]: The [pprof](https://github.com/google/pprof) tool really wants this. If this gets used we might do some intermediate software to manage files.

File: pkg/test/main/README.md
Comment (dougfort) [reply pkg/test/main/README.md:51]: added some test to the README to make it clearer

**Comment by dontlaugh** [comment]
Seems low-risk. Can we merge it? Unless there's some code style stuff we should be aware of.

**Comment by alecholmez** [comment]
@snowp any thoughts on this as a first step?

File: pkg/test/main/main.go
Comment (snowp) [inline pkg/test/main/main.go:159]: Can we use the same pattern here as below where we iterate over the profile types? I would also simplify the output here to just say something like `Turning on block, goroutine and mutex profiling`

**Review by snowp — APPROVED** [review]
Just one non-blocking hit, this LGTM

**Comment by alecholmez** [comment]
Thanks for this @dougfort, just approved the CI process to run so once that passes I'll merge

**Review by alecholmez — APPROVED** [review]

**Pull request merged**
